### PR TITLE
added pytxi

### PR DESCRIPTION
**Pull request by siebrenf**
Added pytxi to convert TPM to gene counts. Pytxi returns gene symbols btw.
Counts table use descriptive>replicate>sample names

todo: 
- ATAC-/ChIP-counts tables need to use descriptive names too
- clustering is failing for ATAC-seq peak counts

**Checklist**
- [x] I made a PR to develop (not master)
- [ ] If applicable: I updated the docs
- [x] I updated the CHANGELOG
- [x] These changes are covered by the tests

### CODE REVIEW

File: seq2science/envs/pytxi.yaml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+name: pytxi
+channels:
+  - bioconda
+  - conda-forge
+  - defaults
+dependencies:
+  - genomepy=0.11.1
+  - pip
+  - pip:
+    - git+https://github.com/vanheeringen-lab/pytxi.git@develop
```
Comment (Maarten-vd-Sande) [inline seq2science/envs/pytxi.yaml:10]: Not sure if this is a permanent solution, but I think we should at least use a tagged version or sth like that..

Comment (siebrenf) [reply seq2science/envs/pytxi.yaml:10]: yea sure! I'm running into some issue during my test run. Once those are fixed it use a commit hash/version tag!

File: seq2science/rules/peak_counts.smk [hunk 2]
```diff
@@ -151,6 +151,13 @@ def get_coverage_table_replicates(file_ext):
 
     return wrapped
 
+def get_names(wildcards):
+    names = [""]
+    for rep in treps[treps["assembly"] == ori_assembly(wildcards.assembly)].index:
+        names.append(rep_to_descriptive(rep, brep=False))
+    names = "\t".join(names)
+    return names
+
```
Comment (Maarten-vd-Sande) [inline seq2science/rules/peak_counts.smk:160]: I double checked, this _should_ be fine. I'm not happy with the way I set this up originally to begin with. Feels like it is **very** easy to make a mistake here... :eyes:

File: seq2science/rules/quantification.smk [hunk 3]
```diff
@@ -31,7 +30,6 @@ if config["quantifier"] == "salmon":
         shell:
             "gffread -w {output} -g {input.fa} {input.gtf} >> {log} 2>&1"
 
-
```
Comment (Maarten-vd-Sande) [inline seq2science/rules/quantification.smk:34]: :eyes:

File: seq2science/scripts/pytxi.py [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+fnames = snakemake.input.cts
+assembly = snakemake.wildcards.assembly
+species = snakemake.input.fa[0]
+from_gtf = snakemake.params.from_gtf
+output = snakemake.output[0]
+sample_names = snakemake.params.names
+
+# redirect all messages to a logfile
+open(logfile, "w")  # start a clean log
+with open(logfile, "a") as log:  # appending because we mix streams (tqdm, logger & std)
```
Comment (Maarten-vd-Sande) [inline seq2science/scripts/pytxi.py:22]: Can't you just do `with open(logfile, "w") as log`? Not sure what happens if you open the file with 'w' and not close it

**Review by Maarten-vd-Sande — APPROVED** [review]

File: seq2science/rules/peak_counts.smk (hunk 2 above)
Comment (siebrenf) [reply seq2science/rules/peak_counts.smk:160]: its the same for loop used to gather the files, so it should always be the correct order. I apply the same method to convert the RNA counts to descriptive elsewhere.

I agree its hacky, so we could leave this out this PR and come back to it later?

File: seq2science/rules/quantification.smk (hunk 3 above)
Comment (siebrenf) [reply seq2science/rules/quantification.smk:34]: copy pasting error i guess.

File: seq2science/scripts/pytxi.py (hunk 4 above)
Comment (siebrenf) [reply seq2science/scripts/pytxi.py:22]: nope, that caused issues for some reason. Loguru should be outputting to stderr, but its clearly more complex.

**Pull request merged**